command/job_start_test.go

```diff
@@ -0,0 +1,245 @@
+	t.Cleanup(func() {
+		for _, jobID := range jobIDs {
+			_ = os.Remove(jobFilePath(jobID))
+		}
+	})
+
+	// record cli output
+	ui := cli.NewMockUi()
+
+	for _, jobID := range jobIDs {
+		job := testJob(jobID)
+
+		jobJSON, err := json.MarshalIndent(job, "", " ")
+		must.NoError(t, err)
+
+		jobFile := jobFilePath(jobID)
+		err = os.WriteFile(jobFile, []byte(jobJSON), 0o644)
+		must.NoError(t, err)
+
+		cmd := &JobRunCommand{Meta: Meta{Ui: ui}}
+
+		code := cmd.Run([]string{"-address", addr, "-json", jobFile})
+		must.Zero(t, code,
+			must.Sprintf("job stop stdout: %s", ui.OutputWriter.String()),
+			must.Sprintf("job stop stderr: %s", ui.ErrorWriter.String()),
+		)
+
+	}
+
+	// helper for stopping a list of jobs
+	stop := func(args ...string) (stdout string, stderr string, code int) {
+		cmd := &JobStopCommand{Meta: Meta{Ui: ui}}
+		code = cmd.Run(args)
+		return ui.OutputWriter.String(), ui.ErrorWriter.String(), code
+	}
+	// helper for starting a list of jobs
+	start := func(args ...string) (stdout string, stderr string, code int) {
+		cmd := &JobStartCommand{Meta: Meta{Ui: ui}}
+		code = cmd.Run(args)
+		return ui.OutputWriter.String(), ui.ErrorWriter.String(), code
+	}
+
+	// stop all jobs in one command
+	args := []string{"-address", addr, "-detach"}
+	args = append(args, jobIDs...)
+	stdout, stderr, code := stop(args...)
+	must.Zero(t, code,
+		must.Sprintf("job stop stdout: %s", stdout),
+		must.Sprintf("job stop stderr: %s", stderr),
+	)
+
+	// start all jobs again in one command
+	stdout, stderr, code = start(args...)
+	must.Zero(t, code,
+		must.Sprintf("job start stdout: %s", stdout),
+		must.Sprintf("job start stderr: %s", stderr),
+	)
+
+}
+
+func TestStartCommand_MultipleCycles(t *testing.T) {
+	ci.Parallel(t)
+
+	srv, client, addr := testServer(t, true, func(c *agent.Config) {
+		c.DevMode = true
+	})
+
+	defer srv.Shutdown()
+
+	ui := cli.NewMockUi()
+
+	job1 := testJob("job-start-test")
+	resp, _, err := client.Jobs().Register(job1, nil)
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	if code := waitForSuccess(ui, client, fullId, t, resp.EvalID); code != 0 {
+		t.Fatalf("status code non zero saw %d", code)
+	}
+
+	args := []string{"-address", addr, "-detach"}
+	args = append(args, "job-start-test")
+	expectedVersions := []uint64{0, 2, 4, 6, 8, 10}
+	stopCmd := &JobStopCommand{Meta: Meta{Ui: ui}}
+	startCmd := &JobStartCommand{Meta: Meta{Ui: ui}}
+
+	// check multiple cycles of starting/stopping a job result in the correct version selected
+	for i := range 6 {
+
+		code := stopCmd.Run(args)
+		must.Zero(t, code,
+			must.Sprintf("job stop stdout: %s", ui.OutputWriter.String()),
+			must.Sprintf("job stop stderr: %s", ui.ErrorWriter.String()),
+		)
+
+		code = startCmd.Run(args)
+		must.Zero(t, code,
+			must.Sprintf("job start stdout: %s", ui.OutputWriter.String()),
+			must.Sprintf("job start stderr: %s", ui.ErrorWriter.String()),
+		)
+		must.Eq(t, expectedVersions[i], startCmd.versionSelected)
```

Comment:
Looks like you've still got a test failure here.

Reply:
Ah I see, I keep passing this test when I run it locally, but it seems to keep failing when I push, could this be because of race conditions I haven't accounted for (similarly to what you mentioned above)?